java/core/src/main/java/sleeper/core/properties/table/TablePropertiesStore.java

```diff
@@ -71,6 +71,16 @@ public TableProperties loadByName(String tableName) {
                 .orElseThrow(() -> TableNotFoundException.withTableName(tableName));
     }
 
+    /**
+     * Checks if a table exists already.
+     *
+     * @param  tableName the table name
+     * @return           true if the table exists, false otherwise
+     */
+    public boolean doesTableExistByName(String tableName) {
+        return tableIndex.getTableByName(tableName).isPresent();
+    }
```

Comment:
This raises a sort of pre-existing problem. The existing save and createTable methods aren't transactional. They decide what to do based on whether the table exists or not, but if another process updates it at the same time then the behaviour is not well defined.

If we ever have a way to make it transactional, the save/createTable give us a place to do that. If we start querying whether the table exists before we call the store, we lose that ability.

We could add an update method that will fail if the table doesn't already exists.